File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    stack<char> stk;
    
    stk.push(s[0]);
    for (int i=1; i<s.size(); i++){
        if (s[i]=='(') stk.push(s[i]);
        else{
            if (!stk.empty()) stk.pop();
            else return false;
        }
    }
    
    if (stk.empty()) return true;
    else return false;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):    
    stk = deque()
    
    for c in s:
        if c == '(':
            stk.append(c)
            
        if c == ')':
            try:
                stk.pop()
            except Exception:
                return False
                
    return len(stk) == 0
                
                
